static/js/components/AddSourceGroup.jsx

```diff
@@ -0,0 +1,174 @@
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import { useDispatch } from "react-redux";
+import { useForm, Controller } from "react-hook-form";
+
+import Dialog from "@material-ui/core/Dialog";
+import DialogContent from "@material-ui/core/DialogContent";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import Checkbox from "@material-ui/core/Checkbox";
+import Button from "@material-ui/core/Button";
+import IconButton from "@material-ui/core/IconButton";
+import AddCircleIcon from "@material-ui/icons/AddCircle";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import Tooltip from "@material-ui/core/Tooltip";
+import { makeStyles } from "@material-ui/core/styles";
+
+import { showNotification } from "baselayer/components/Notifications";
+import * as sourceActions from "../ducks/source";
+import FormValidationError from "./FormValidationError";
+
+const useStyles = makeStyles(() => ({
+  iconButton: {
+    display: "inline-block",
+  },
+}));
+
+const displayStyle = (isDisplayed) => ({
+  display: isDisplayed ? "inline-block" : "none",
+});
+
+const AddSourceGroup = ({ source, userGroups, icon }) => {
+  const classes = useStyles();
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  // Dialog logic:
+
+  const dispatch = useDispatch();
+  const [dialogOpen, setDialogOpen] = useState(false);
+
+  const { handleSubmit, errors, reset, control, getValues } = useForm();
+
+  useEffect(() => {
```

Comment:
As this file looks to be a slightly modified version of SaveCandidateButton.jsx, I'm wondering if there's a way to generalize that one enough to be used both places so as to not have so much duplicated code. Thoughts?

Reply:
Good point. It might be a good idea to factor out the dialog form for selecting and submitting the new group. That's the part that's most duplicated, and I believe Dima uses the same logic in the Alerts page on Fritz for saving directly from the alert as well.  Should I do that as part of this PR?

Reply:
Let's defer this to another PR